src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res

```diff
@@ -0,0 +1,534 @@
+
+      switch paypalDict->Js.Json.classify {
+      | JSONString(str) => setSetupAccountStatus(._ => str->PayPalFlowUtils.stringToVariantMapper)
+      | JSONObject(dict) =>
+        handleObjectResponse(
+          ~dict,
+          ~setSetupAccountStatus,
+          ~setInitialValues,
+          ~connector,
+          ~handleStateToNextPage,
+        )
+      | _ => ()
+      }
+      setScreenState(_ => PageLoaderWrapper.Success)
+    } catch {
+    | _ => setScreenState(_ => PageLoaderWrapper.Error(""))
+    }
+  }
+
+  React.useEffect0(() => {
+    if isRedirectedFromPaypalModal {
+      getStatus()->ignore
+    }
+    if !isUpdateFlow {
+      RescriptReactRouter.replace("/connectors/new?name=paypal")
+    }
+    setSetupAccountStatus(._ => Account_not_found)
+    None
+  })
+
+  let validateMandatoryFieldForPaypal = values => {
+    let errors = Dict.make()
+    let valuesFlattenJson = values->JsonFlattenUtils.flattenObject(true)
+    let profileId = valuesFlattenJson->getString("profile_id", "")
+    if profileId->Js.String2.length === 0 {
+      Dict.set(errors, "Profile Id", `Please select your business profile`->Js.Json.string)
+    }
+    errors->Js.Json.object_
+  }
+
+  let handleConnector = async values => {
+    try {
+      await onSubmitMain(values)
+      setSetupAccountStatus(._ => Redirecting_to_paypal)
+    } catch {
+    | Js.Exn.Error(_e) => ()
+    }
+  }
+
+  let handleOnSubmit = (values, _) => {
+    switch setupAccountStatus {
+    | Account_not_found =>
+      switch configuartionType {
+      | Manual
+      | NotSelected => {
+          let authType =
+            initialValues
+            ->getDictFromJsonObject
+            ->getDictfromDict("connector_account_details")
+            ->getString("auth_type", "")
+            ->Js.String2.toLowerCase
+            ->ConnectorUtils.mapAuthType
+
+          let temporaryAuthDict =
+            [("auth_type", "TemporaryAuth"->Js.Json.string)]->getJsonFromArrayOfJson
+
+          let dictOfInitialValues = values->getDictFromJsonObject
+
+          setSetupAccountStatus(._ => Manual_setup_flow)
+          if isUpdateFlow && authType === #SignatureKey {
+            dictOfInitialValues->Dict.set("connector_account_details", temporaryAuthDict)
+            setInitialValues(_ => dictOfInitialValues->Js.Json.object_)
+          }
+        }
+
+      | Automatic => handleConnector(values)->ignore
+      }
+    | Manual_setup_flow => {
+        let dictOfInitialValues = values->getDictFromJsonObject
+        dictOfInitialValues->Dict.set("disabled", false->Js.Json.boolean)
+        dictOfInitialValues->Dict.set("status", "active"->Js.Json.string)
+        setInitialValues(_ => dictOfInitialValues->Js.Json.object_)
+        handleConnectorConnected(dictOfInitialValues->Js.Json.object_)->ignore
+      }
+    | _ => ()
+    }
+    Js.Nullable.null->Js.Promise.resolve
+  }
+
+  let proceedButton = switch setupAccountStatus {
```

Comment:
`<UIUtils.RenderIf condition={setupAccountStatus !==Redirecting_to_paypal}>
 <FormRenderer.SubmitButton
      loadingText="Processing..."
      text="Proceed"
      disabledParamter={configuartionType === NotSelected ? true : false}
    />

</UIUtils.RenderIf>

`


Directly inside Form can be written

Reply:
Wrapped the component with UIUtils than switch case